File: leetcode/solutions/0001.TwoSum/twoSum.ts
export default function twoSum(
  nums: number[],
  target: number
): [number, number] | null {
  // [number, index of the number in nums]
  const map = new Map<number, number>();

  for (const [i, num] of nums.entries()) {
    const pair = map.get(target - num);

    if (pair != null) {
      return [pair, i];
    }

    map.set(num, i);
  }

  return null;
}

File: leetcode/solutions/0001.TwoSum/twoSum.go
package solution

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		if j, ok := numMap[target-num]; ok {
			return []int{i, j}
		}

		numMap[num] = i
	}

	return nil
}
